// beautiful-console/src/color/index.ts
import {Color} from '../type'

export const gray: Color = [
    '#f8f9fa',
    '#ebedf0',
    '#dee0e5',
    '#cfd3d9',
    '#bec4cc',
    '#acb3be',
    '#97a0ae',
    '#7e899a',
    '#626c7d',
    '#393f49'
]

export const blue: Color = [
    '#e6effd',
    '#cbddfb',
    '#adcaf8',
    '#88b2f5',
    '#5a94f2',
    '#1467ed',
    '#125cd5',
    '#0f51ba',
    '#0c4299',
    '#092f6c'
]

export const indigo: Color = [
    '#edeafd',
    '#d8d4fb',
    '#c1baf9',
    '#a59af7',
    '#7f6ff4',
    '#2d14ed',
    '#2812d5',
    '#230fba',
    '#1d0c99',
    '#14096b'
]

export const violet: Color = [
    '#f5eafd',
    '#ebd2fb',
    '#e0b7f9',
    '#d196f6',
    '#be6af3',
    '#9a14ed',
    '#8b12d6',
    '#790fbb',
    '#640d9a',
    '#47096e'
]

export const fuchsia: Color = [
    '#fdeafb',
    '#fbd3f7',
    '#f9b9f2',
    '#f799ec',
    '#f36ee4',
    '#ed14d4',
    '#d612bf',
    '#bc0fa8',
    '#9c0d8b',
    '#700965'
]

export const pink: Color = [
    '#fdeaf1',
    '#fbd3e3',
    '#f9b9d1',
    '#f799bd',
    '#f36da1',
    '#ed1467',
    '#d6125d',
    '#bc0f51',
    '#9c0d43',
    '#700930'
]

export const red: Color = [
    '#fdebe9',
    '#fbd6d1',
    '#f9bdb5',
    '#f69e93',
    '#f37666',
    '#ed2d14',
    '#d62812',
    '#bb230f',
    '#9b1d0d',
    '#6f1509'
]

export const orange: Color = [
    '#fcf2e2',
    '#fae5c2',
    '#f7d69f',
    '#f4c578',
    '#f1b14b',
    '#ed9a14',
    '#d68b12',
    '#bc7a0f',
    '#9b650d',
    '#704809'
]

export const yellow: Color = [
    '#f9fce1',
    '#f3fac0',
    '#edf79d',
    '#e5f475',
    '#ddf148',
    '#d4ed14',
    '#c0d612',
    '#a8bc0f',
    '#8c9d0d',
    '#667209'
]

export const lime: Color = [
    '#edfce4',
    '#dafac6',
    '#c4f8a4',
    '#abf57e',
    '#8df150',
    '#67ed14',
    '#5dd612',
    '#52bc0f',
    '#449d0d',
    '#317109'
]

export const green: Color = [
    '#e7fdea',
    '#cdfbd3',
    '#b0f8b8',
    '#8cf698',
    '#5df26f',
    '#14ed2d',
    '#12d628',
    '#0fbc23',
    '#0d9d1d',
    '#097215'
]

export const teal: Color = [
    '#e6fdf4',
    '#cbfbe8',
    '#acf8db',
    '#87f5cb',
    '#58f2b7',
    '#14ed9a',
    '#12d68b',
    '#0fbc7a',
    '#0d9d66',
    '#09724a'
]

export const cyan: Color = [
    '#e4fafc',
    '#c7f4fa',
    '#a6eef8',
    '#80e7f5',
    '#51dff1',
    '#14d4ed',
    '#12bfd6',
    '#0fa8bc',
    '#0d8c9c',
    '#096571'
]

export default {
    gray,
    blue,
    indigo,
    violet,
    fuchsia,
    pink,
    red,
    orange,
    yellow,
    lime,
    green,
    teal,
    cyan
}
